enhancements/upgrade-readiness.md

```diff
@@ -0,0 +1,540 @@
+
+| Field | Description |
+| ----- | ----------- |
+| Type | Type of condition in CamelCase or in foo.example.com/CamelCase. |
+| Status | Status of the condition, one of True, False, Unknown. |
+| ObservedGeneration | If set, this represents the .metadata.generation that the condition was set based upon. |
+| LastTransitionTime | Last time the condition transitioned from one status to another. |
+| Reason | The reason for the condition's last transition in CamelCase. |
+| Message | A human readable message indicating details about the transition. |
+
+OLM will define a set of "OLM Supported Conditions" which will influence the behavior of OLM when present. The list of "OLM Supported Conditions" can be seen in the table below:
+
+| Condition Type | Effect |
+| -------------- | ------ |
+| Upgradeable | When `false`, OLM will not upgrade the operator.|
+| Important | OLM will highlight this condition to the Cluster Admin.|
+
+Operators will communicate their conditions through one of two methods:
+
+1. By setting conditions on CustomResource that the operator owns.
+2. By setting conditions on a `Probe` CustomResource directly.
+
+### Why allow operators to communicate status through CustomResources
+
+There are many reasons why operators should communicate their conditions through owned CustomResources:
+
+- CustomResources are the main entry point to an operator and operators are already expected to update the status of the CustomResource to communicate "Actual State".
+- By writing the condition to the status of an owned CustomResource, it becomes very clear which CR is responsible for changing OLM behavior.
+- Introducing an OLM owned CRD forces operators to integrate with OLM at a much deeper level than writing to a resource that they already manage. Operators would be required to detect if it is being managed by OLM and begin writing to the API. By writing to an owned CR, the Operator does not require OLM specific code and can be run independently of OLM.
+
+Allowing operators to write to the status of any CR creates a potential scaling issue if enough CRs are on cluster. OLM will avoid these performance issues by introducing the Status Probe Operator (SPO) which will reconcile the `probe` resource. If an operator chooses to write to their own CRs, SPO will create a pod that will keep the status of the `probe` resource synced with the conditions in the CRs.
+
+### Opting into OLM Supported Conditions
+
+Operators may opt-in to "OLM Supported Conditions" by adding annotations to their owned CRD(s), where the key is the defined by OLM and the value is a boolean expression over condition types applied to the CR at `.status.conditions`. The "OLM Supported Condition" annotation key will adhere to the following format: `olm.Condition.<ConditionName>`.
```

Comment:
could this be `operators.coreos.com/condition.<condition-name>` instead?

Reply:
Absolutely.

Reply:
Addressed in the latest version of the PR.